src/include/migraphx/op/slice.hpp

```diff
@@ -120,15 +142,33 @@ struct slice
             auto axis = axes[i];
             new_lens[axis] =
                 fix_index(old_lens, axis, ends[i]) - fix_index(old_lens, axis, starts[i]);
+            if(input_shape.dynamic())
+            {
+                new_mins[axis] = std::min(new_lens[axis], new_mins[axis]);
+                if(new_opts[axis] < new_mins[axis] or new_opts[axis] > new_lens[axis])
+                    new_opts[axis] = 0;
+            }
+        }
+        if(input_shape.dynamic())
+        {
+            return shape{t, new_mins, new_lens, new_opts};
+        }
+        else
+        {
+            return shape{t, new_lens, old_strides};
         }
-        return shape{t, new_lens, old_strides};
     }
 
-    argument compute(shape output_shape, std::vector<argument> args) const
+    argument compute(const dyn_output& dyn_out, std::vector<argument> args) const
     {
-        auto input  = args[0];
-        auto offset = compute_offset(input.get_shape()) * output_shape.type_size();
-        return {std::move(output_shape), [=] { return input.data() + offset; }};
+        auto input = args[0];
+        if(dyn_out.computed_shape.dynamic())
+            return {dyn_out.computed_shape, [=] { return input.data(); }};
```

Comment:
this  line is not covered by codecov, Do we have Ref test to cover this ?

Reply:
That entire _if_ block shouldn't even be there--removing.